.github/workflows/template-tryvi-analysis.yaml

```
@@ -0,0 +1,189 @@
+name: Tryvi Analysis
+
+on:
+  workflow_call:
+    inputs:
+      CLUSTER_ENV:
+          required: false
+          type: string
+          default: prod
+          description: Cluster environment (ex. prod)
+      CLUSTER_LOCATION:
+          required: false
+          type: string
+          default: northamerica
+          description: Cluster environment (ex. northamerica, europe)
+      ZONE:
+          required: false
+          type: string
+          default: northamerica-northeast1-a
+          description: Cluster location (ex. northamerica-northeast1-a)
+      RUNNER:
+          required: false
+          type: string
+          default: dm-prod-na
+          description: Github runner name
+      REF:
+          required: false
+          type: string
+          default: ${{ github.ref }}
+          description: Branch/Commit/Tag to checkout
+      KUBECTL_VERSION:
+          required: false
+          type: string
+          default: v1.27.6
+          description: Kubectl version to use
+      SEVERITY:
+          required: false
+          type: string
+          default: HIGH,CRITICAL
+          description: Severity to fail on
+      FILTERS:
+          required: false
+          type: string
+          default: ''
+          description: Filter images to scan
+    secrets:
+      GCP_SA_KEY:
+          required: true
+          description: Google cloud github action service account key
+      GH_CI_CD_TOKEN:
+          required: true
+          description: GitHub - De Marque CI/CD aka Jenkins personnal access tokens (classic)
+      SLACK_WEBHOOK_URL:
+          required: true
+          description: Product slack webhook url
+
+permissions: read-all
+
+jobs:
+    fetch_images_path:
+      name: Fetch Docker Images Path
+      runs-on: ${{ inputs.RUNNER }}
+
+      steps:
+        - name: Checkout Repository
+          uses: actions/checkout@v4.1.1
+          with:
+            ref: ${{ inputs.REF }}
+
+        - name: GCloud Authentification
+          uses: google-github-actions/auth@v1.1.1
+          with:
+            credentials_json: ${{ secrets.GCP_SA_KEY }}
+
+        - name: Prepare Credentials
+          run: |
+            mkdir -p creds
+            ln -s $GOOGLE_APPLICATION_CREDENTIALS creds/serviceaccount.json
+
+        - name: Get Cluster Credentials
+          uses: google-github-actions/get-gke-credentials@v2.0.0
+          with:
+            cluster_name: ${{ inputs.CLUSTER_ENV }}-${{ inputs.CLUSTER_LOCATION }}
+            location: ${{ inputs.ZONE }}
+
+        - name: Set Up Kubectl
+          uses: azure/setup-kubectl@v3
+          with:
+            version: ${{ inputs.KUBECTL_VERSION }}
+
+        - name: Fetch Images
+          run: |
+            if [ -n "${{ inputs.FILTERS }}" ]; then
+                kubectl get deploy -A -o jsonpath='{range .items[*]}{.spec.template.spec.containers[0].image}{"\n"}' | sort -u | grep "${{ inputs.FILTERS }}" > docker_images_path.txt
```

Je crois bien que tu pourrais enlever le `if` et juste faire ca:
```suggestion
                kubectl get deploy -A -o jsonpath='{range .items[*]}{.spec.template.spec.containers[0].image}{"\n"}' | sort -u | grep "${{ inputs.FILTERS }}.*" > docker_images_path.txt
```

---

```
~ $ echo -e "nginx\ncoco"
nginx
coco
~ $ echo $FILTERS

~ $ echo -e "nginx\ncoco" | grep "${FILTERS}.*"
nginx
coco
~ $ FILTERS=nginx
~ $ echo -e "nginx\ncoco" | grep "${FILTERS}.*"
nginx
```